### PR TITLE
Flask Templates for Grid Routing

### LINKED ISSUE #723: Create Flask Composition Templates to work with Grid Routing
We need to create templates for grid data and routes for a generated flask application to be able to render grid route.

### LINKED ISSUE #724: Create Flask Composition Templates to work with Master Detail Routing
We need to create templates for master detail data and routes for a generated flask application to be able to render master detail route.

### PR DESCRIPTION
**Pull request by streetcoder123**
Task: https://microsoftgarage.visualstudio.com/Intern%20GitHub/_workitems/edit/27727

This PR adds grid Routing support using Sample Data to the Flask Application.

**How to Check**
- Generate a Flask Application
- Go the application directory and run `yarn install`
- Run 'yarn start` to run the application

- Go to /grid route to see if it is working

closes #723

### CODE REVIEW

File: templates/Web/Projects/FlaskDefault/server/constants.py [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import os
 
-CONSTANTS = {
-    'PORT': os.environ.get('PORT', 3001),
-}
+CONSTANTS = {}
+    
+CONSTANTS['PORT'] = os.environ.get('PORT', 3001)
```
Comment (piotrmark) [inline templates/Web/Projects/FlaskDefault/server/constants.py:5]: Why? :)

File: templates/Web/_composition/Flask/Page.Flask.Grid.AddRoutes/Page.Flask.Grid.AddRoutes/Page.Flask.Grid.AddRoutes/server/constants_postaction.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import os
+
+CONSTANTS = {}
+
+//{[{
+CONSTANTS['ENDPOINT'] = {}
+//}]}
+//{[{
+CONSTANTS['ENDPOINT']['GRID'] = '/api/grid'
+//}]}
```
Comment (piotrmark) [inline templates/Web/_composition/Flask/Page.Flask.Grid.AddRoutes/Page.Flask.Grid.AddRoutes/Page.Flask.Grid.AddRoutes/server/constants_postaction.py]: Ok, I know why :D

**Review by piotrmark — DISMISSED** [review]
Looks good

File: templates/Web/_composition/Flask/Page.Flask.AddSampleData.ForAllExceptList/Page.Flask.AddSampleData.ForAllExceptList/server/sampleData.py [hunk 3]
```diff
@@ -0,0 +1,5 @@
+sampleData = {}
+
+# This class holds sample data used by some generated pages to show how they can be used.
+# Todo Web Template Studio: Delete this file once your app is using real data.
```
Comment (jasmingoh) [inline templates/Web/_composition/Flask/Page.Flask.AddSampleData.ForAllExceptList/Page.Flask.AddSampleData.ForAllExceptList/server/sampleData.py]: I am nitpicking here, "Todo" is all capitalized for the node template. Is "Todo" the convention in python?

File: templates/Web/_composition/Flask/Page.Flask.Grid.AddRoutes/Page.Flask.Grid.AddRoutes/Page.Flask.Grid.AddRoutes/server/constants_postaction.py [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import os
+
+CONSTANTS = {}
+
+//{[{
+CONSTANTS['ENDPOINT'] = {}
+//}]}
+//{[{
+CONSTANTS['ENDPOINT']['GRID'] = '/api/grid'
```
Comment (jasmingoh) [inline templates/Web/_composition/Flask/Page.Flask.Grid.AddRoutes/Page.Flask.Grid.AddRoutes/Page.Flask.Grid.AddRoutes/server/constants_postaction.py]: I am comparing this with the node template. Why is it '/api/grid'? It is only '/grid' in the node template.

link to node: https://github.com/microsoft/WebTemplateStudio/blob/dev/templates/Web/_composition/NodeJS/Page.Node.Grid.AddRoutes/server/constants_postaction.js

**Comment by jasmingoh** [comment]
Question about the structure of the Flask folder, why are there multiple layers with the same name? 

image

File: templates/Web/_composition/Flask/Page.Flask.Grid.AddRoutes/Page.Flask.Grid.AddRoutes/Page.Flask.Grid.AddRoutes/server/constants_postaction.py (hunk 4 above)
Comment (streetcoder123) [reply templates/Web/_composition/Flask/Page.Flask.Grid.AddRoutes/Page.Flask.Grid.AddRoutes/Page.Flask.Grid.AddRoutes/server/constants_postaction.py]: In Node, '/api' is appended to every route for the routes file. It is essentially, '/api/route' for every route that front-end makes request for.

File: templates/Web/_composition/Flask/Page.Flask.AddSampleData.ForAllExceptList/Page.Flask.AddSampleData.ForAllExceptList/server/sampleData.py (hunk 3 above)
Comment (streetcoder123) [reply templates/Web/_composition/Flask/Page.Flask.AddSampleData.ForAllExceptList/Page.Flask.AddSampleData.ForAllExceptList/server/sampleData.py]: Yeah, it should be capitalized. Corrected it

**Comment by streetcoder123** [comment]
@jasmingoh Good catch on the folder structure. I don't know how I created nested folder structure for this. I have corrected that.

**Review by jasmingoh — APPROVED** [review]
Looks good!

**Review by jcoc611-microsoft — APPROVED** [review]

**Review by annielovesjs — APPROVED** [review]
Works well!

**Pull request merged**